sn_interface/src/network_knowledge/section_tree/mod.rs

```diff
@@ -225,62 +247,45 @@ impl SectionTree {
                 // We are not aware of the prefix, let's then verify it can be
                 // trusted based on our own section chain and the provided proof chain.
-                if !proof_chain.check_trust(self.sections_dag.keys()) {
-                    return Err(Error::UntrustedProofChain(format!(
-                        "none of the keys were found on our section chain: {:?}",
+                if !partial_dag.check_trust(self.sections_dag.keys()) {
+                    return Err(Error::UntrustedPartialDAG(format!(
+                        "None of the keys were found on our section chain: {:?}",
                         signed_sap.value
                     )));
                 }
             }
         }
 
-        // Make sure the proof chain can be trusted,
-        // i.e. check each key is signed by its parent/predecessor key.
-        if !proof_chain.self_verify() {
-            return Err(Error::UntrustedProofChain(format!(
-                "invalid proof chain: {:?}",
-                proof_chain
-            )));
-        }
-
-        // Check the SAP's key is the last key of the proof chain
-        if *proof_chain.last_key() != signed_sap.section_key() {
-            return Err(Error::UntrustedSectionAuthProvider(format!(
-                "section key ({:?}, from prefix {:?}) isn't in the last key in the proof chain provided. (Which ends with ({:?}))",
-                signed_sap.section_key(),
-                signed_sap.prefix(),
-                proof_chain.last_key()
-            )));
-        }
-
         // We can now update our knowledge of the remote section's SAP.
         // Note: we don't expect the same SAP to be found in our records
         // for the prefix since we've already checked that above.
-        let changed = self.insert(signed_sap);
-
-        // update our sections_dag with the proof_chain. Cannot be an error, since in cases where we
-        // have outdated SAP (aware of prefix)/ not aware of the prefix, we have the proof_chains's
-        // root/child key in our sections_dag. Checked in the above match statement.
-        self.sections_dag.join(proof_chain.clone())?;
-
-        for (prefix, section_key) in &self.sections {
-            debug!(
-                "Known prefix,section_key after update: {:?} = {:?}",
-                prefix, section_key
-            );
+        if self.insert(signed_sap) {
+            // update our sections_dag with the proof_chain. Cannot be an error, since in cases where we
+            // have outdated SAP (aware of prefix)/ not aware of the prefix, we have the proof_chains's
+            // root/child key in our sections_dag. Checked in the above match statement.
+            self.sections_dag.merge(partial_dag.clone())?;
+            for (prefix, section_key) in &self.sections {
+                debug!(
+                    "Known prefix,section_key after update: {:?} = {:?}",
+                    prefix, section_key
+                );
```

Comment:
nit. save some vertical space
```suggestion
                debug!("Known prefix, section_key after update: {prefix:?} = {section_key:?}");
```